### ISSUE TITLE
A Sample which is More Simple

### ISSUE THREAD

**Issue by DavidRogersDev** [urfnet/URF.Core#87]
I have always felt that the sample code for this project is too complex. There are too many dependencies.
To name 1 at a pinch, OData. This is unnecessary to demonstrate a Unit of Work/Repository implementation.
(I've never actually worked on a project which uses OData, so I'd have to go and figure out how to create Postman requests to simulate an OData request.)

It took me about 2 hours to get the Sample code building today.
In fact, I never actually got it to build. So, I took the following projects in the "sample" repo and got them working in a fresh ASPNETCORE API project/solution:

- Northwind.Data
- Northwind.Entity
- Northwind.Repository
- Northwind.Service

No OData, no Angular (I have no Angular experience) and I was able to use Postman to test the code.

In my opinion, users should have a nice "F5 experience".
One should endeavor to make it so users can simply clone, build and then play.

In a nutshell, minimal dependencies are preferred. There's already enough fiddling, getting the right version of .NET installed, or upgrading to the current, as the case may be.
Otherwise, people may give up after the 1st 45 minutes of trying to get all the right dependencies on their PC etc.

Even a console application would do the trick. The generic host makes it easy to add DI etc.

Perhaps you could keep your existing Sample as well for those users who want a full, end-to-end example.
But a simple, basic sample repo would be very helpful. I'm willing to take a crack at creating such a repo, if it helps.

This was meant as constructive feedback and as a way of opening a dialogue about this issue. Cheers.

**Comment by tonysneed** [comment]
Have you looked at [URF.Core.Sample.v3](https://github.com/urfnet/URF.Core.Sample.v3)?

**Comment by DavidRogersDev** [comment]
Hi Tony,
I found that last night. It certainly is a lot easier to get going that the other sample.
It also emphasizes a different use of the Project. Direct use of the IRepository, rather than inheriting from the Service class.

I plan to create a bunch of small samples for myself, as I play with the latest version and get on top of its various features.
I'll make a repo out of that and if you find it useful, you are most welcome to it.
There's a lot of features in URF and it seems to be one of those things where you only use what you need.

While I'm here, I like the way Trackable Entities has been integrated into this. I love Trackable Entities and have found it to be so good if you want to manage the disconnected state in that manner (that is - paint the state from the client).

By all means, close this issue. Perhaps I'll ping you when I get a few samples going in a repo.
